### PR TITLE
Updating htpasswd logic to not mount a volume for creating file

### PR DESCRIPTION
**Pull request by ewolinetz**
Addresses #4046 

cc @mwoodson

### CODE REVIEW

File: roles/openshift_metrics/tasks/generate_hawkular_certificates.yaml [hunk 1]
```diff
@@ -40,12 +40,19 @@
...
 
-- name: generate htpasswd file for hawkular metrics
+- name: generate htpasswd content for hawkular metrics
   shell: >
-    docker run -i -a STDIN -a STDOUT -v "{{ mktemp.stdout }}"
+    cat "{{ mktemp.stdout}}/hawkular-metrics.pwd" | cat > metrics.pwd && docker run -i -a STDIN -a STDOUT
     "{{ openshift_metrics_httpd_image }}" htpasswd -ni hawkular
-    > '{{ mktemp.stdout }}/hawkular-metrics.htpasswd'
-    < '{{ mktemp.stdout }}/hawkular-metrics.pwd'
+    < metrics.pwd
```
Comment (sdodson) [inline roles/openshift_metrics/tasks/generate_hawkular_certificates.yaml]: Why not just the following avoiding cat from tmp to pwd?

```
docker run -i -a STDIN -a STDOUT
"{{ openshift_metrics_httpd_image }}" htpasswd -ni hawkular
< "{{ mktemp.stdout}}/hawkular-metrics.pwd"
```

If we stick with this as is do we need to clean up metrics.pwd?

Comment (ewolinetz) [inline roles/openshift_metrics/tasks/generate_hawkular_certificates.yaml]: i'll double check that i wasn't doing something else odd when i was testing that... but IIRC it wouldn't validate doing it that way. let me confirm

Comment (ewolinetz) [inline roles/openshift_metrics/tasks/generate_hawkular_certificates.yaml]: yeah, i must have been testing it incorrectly. it validated using the approach you pasted above

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by sdodson** [comment]
Tested locally, CI won't test this. Merging.

**Pull request merged**